improver/utilities/field_texture.py

```diff
@@ -0,0 +1,249 @@
+            MANDATORY_ATTRIBUTE_DEFAULTS,
+            data=ratio,
+        )
+        return ratio
+
+    @staticmethod
+    def _calculate_transitions(data):
+        """
+        Identifies actual transitions present in a binary field. These transitions
+        are defined as the number of cells of value zero that directly neighbour
+        cells of value 1. The number of transitions is calculated for all cells
+        of value 1. The number of transitions for cells of value 0 is set to 0.
+
+        Args:
+            data (numpy.ndarray):
+                A NumPy array of the input cube for data manipulation.
+
+        Returns:
+            numpy.ndarray:
+                A NumPy array containing the transitions for ratio calculation.
+        """
+        padded_data = np.pad(data, 1, mode="edge")
+        diff_x = np.abs(np.diff(padded_data, axis=1))
+        diff_y = np.abs(np.diff(padded_data, axis=0))
+        cell_sum_x = diff_x[:, 0:-1] + diff_x[:, 1:]
+        cell_sum_y = diff_y[0:-1, :] + diff_y[1:, :]
+        cell_sum = cell_sum_x[1:-1, :] + cell_sum_y[:, 1:-1]
+        cell_sum = np.where(data > 0, cell_sum, 0)
+        return cell_sum
+
+    def process(self, input_cube):
+        """
+        Calculates a field of texture to use in differentiating solid and
+        more scattered features.
+
+        Args:
+            input_cube (iris.cube.Cube):
+                Input data in cube format containing the field for which the
+                texture is to be assessed.
+
+        Returns:
+            iris.cube.Cube:
+                A cube containing the mean of the thresholded ratios in cube
+                format.
+        """
+
+        values = np.unique(input_cube.data)
+        non_binary = np.where((values != 0) & (values != 1), True, False)
+        if non_binary.any():
+            raise ValueError("Incorrect input. Cube should hold binary data only")
+
+        # Create new cube name for _calculate_ratio method.
+        self.cube_name = find_threshold_coordinate(input_cube).name()
+
+        # Extract threshold from input data to work with.
+        cube = input_cube.extract(
+            iris.Constraint(
+                coord_values={
+                    self.cube_name: lambda cell: cell == self.diagnostic_threshold
+                }
+            )
+        )
+        try:
+            cube.remove_coord(self.cube_name)
+        except AttributeError:
+            msg = "Threshold {} is not present on coordinate with values {} {}"
+            raise ValueError(
+                msg.format(
+                    self.diagnostic_threshold,
+                    input_cube.coord(self.cube_name).points,
+                    input_cube.coord(self.cube_name).units,
+                )
+            )
+        ratios = iris.cube.CubeList()
+
+        try:
+            cslices = cube.slices_over("realization")
+        except CoordinateNotFoundError:
+            cslices = [cube]
+        for cslice in cslices:
```

Comment:
Please could you add a line break above this line, for readability.

Reply:
clarification: below try:? 
is this convention, and if so, should I do this in the above try statement too?

Reply:
No: above `for cslice in cslices`.  The `try / except` clause is a self-contained statement; the `for` loop is then the next step so there should be a blank line above it.

Reply:
Done